src/plugins/discourse/mirror.test.js

```diff
@@ -467,6 +467,24 @@ describe("plugins/discourse/mirror", () => {
     });
   });
 
+  it("warns if a user's likes are missing", async () => {
+    const {mirror, fetcher, reporter} = example();
+    const pid = fetcher.addPost(1, null, "credbot");
+    (fetcher: any).likesByUser = async () => {
+      throw new Error("404 - Likes not found");
+    };
+    await mirror.update(reporter);
+    expect(mirror.topics()).toEqual([fetcher._topic(1)]);
+    expect(mirror.posts()).toEqual([fetcher._post(pid)]);
+    expect(mirror.likes()).toEqual([]);
+    expect(console.warn).toHaveBeenCalledWith(
+      "Warning: Encountered error '404 - Likes not found' " +
+        "while processing likes for credbot; skipping this user."
+    );
+    expect(console.warn).toHaveBeenCalledTimes(1);
+    jest.spyOn(console, "warn").mockImplementation(() => {});
```

Comment:
Willing to let this slide because I see that it’s following an existing
pattern, but this doesn’t feel right—this should be handled by
`beforeEach`/etc. callbacks. If one of the assertions in the test fails,
we’ll never effect this restoration. But I don’t really understand why
the `beforeEach`/etc. callbacks are set up the way that they are, so
willing to defer.